lib/oauth2/access_token.ex

```diff
@@ -109,170 +104,20 @@ defmodule OAuth2.AccessToken do
   ```
   """
   def new(response, client) do
-    {std, other} = Dict.split(response, @standard)
-
-    struct __MODULE__, [
-      access_token:  std["access_token"],
-      refresh_token: std["refresh_token"],
-      expires_at:    (std["expires_in"] || other["expires"]) |> expires_at(),
-      token_type:    std["token_type"] |> normalize_token_type(),
-      other_params:  other,
-      client:        remove_headers(client)]
+    {std, other} = Map.split(response, @standard)
+
+    token =
+      struct(__MODULE__, [
+        access_token:  std["access_token"],
+        refresh_token: std["refresh_token"],
+        expires_at:    (std["expires_in"] || other["expires"]) |> expires_at(),
+        token_type:    std["token_type"] |> normalize_token_type(),
+        other_params:  other
+      ])
+    %{client | headers: [], token: token}
```

Comment:
why have empty headers? Shouldn't there be a way to set these? For e.g. Slack is really into using form encoded requests rather than JSON. It'd kinda suck to have to keep specifying this after you created a new one.